src/web/mjs/connectors/VizShonenJump.mjs

```diff
@@ -0,0 +1,200 @@
+                id: manga.pathname,
+                title: manga.querySelector(':nth-child(2)').innerText.trim()
+            };
+        });
+    }
+
+    async _getChapters(manga) {
+        const request = new Request(new URL(manga.id, this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, 'a.o_chapter-container');
+        data = data.filter(chapter => chapter.innerText.includes('FREE'));
+
+        return data.map(chapter => {
+            try {
+                return {
+                    id: chapter.href,
+                    title: chapter.querySelector('.disp-id').innerText.trim()
+                };
+            } catch(error) {
+                try {
+                    return {
+                        id: chapter.href,
+                        title: 'Ch. ' + chapter.href.match(/chapter-(\d+)\//)[1]
+                    };
+                } catch(error) {
+                    throw new Error('Unknown chapter format. Please report at https://github.com/manga-download/hakuneko/issues');
+                }
+            }
+        });
+    }
+
+    async _getPages(chapter) {
+        let script = `
+            new Promise((resolve, reject) => {
+                setTimeout(() => {
+                    try {
+                        window.loadPages = function loadPages() {
+                            var aj_images = [];
+                            for (var page_count = 0; page_count <= pages; page_count++) 
+                                if (!(page_count < 0 || page_count > pages) && "undefined" == typeof pageImages["page" + page_count]) {
+                                    var aj_req = {
+                                        url: pageUrl + "&manga_id=" + manga_id + "&page=" + page_count,
+                                        dataType: "text"
+                                    };
+                                    aj_images.push(aj_req)
+                                } 
+                                resolve(Object.values(aj_images));
+                        }
+
+                        window.loadPages();
+
+                    } catch(error) {
+                        reject(error);
+                    }
+                }, 2000);
+            });
+        `;
+        let request = new Request(new URL(chapter.id, this.url));
+        let data = await Engine.Request.fetchUI(request, script, 60000, false);
+
+        return data.map(page => this.createConnectorURI(
+            {
+                id: page.url,
+                referer: chapter.id
+            }
+        ));
+    }
+
+    async _handleConnectorURI(payload) {
+        while (this.concurrent >= this.config.concurrent.value) {
+            await this.throttle_connections( Math.floor(Math.random() * (2000 - 1000)) + 1000);
+        }
+        this.concurrent += 1;
+
+        try {
+            // API request
+            let request = new Request(new URL(payload.id, new URL(this.url).origin ), this.requestOptions);
```

Comment:
Just use `this.url` instead of `new URL(this.url).origin`, the URL constructor will magically resolve the path correctly